### ISSUE TITLE
Add DisplayName in custom attribute

### ISSUE THREAD

**Issue by adamshakhabov** [xunit/xunit#2635]
## Suggestion

It would be good to be able to set DisplayName for test cases that defined via a custom attribute. For example, it can be a new `DataAtrribute.DisplayName` property which can be used in the following way:

```C#
public class MyCustomDataAttribute : DataAttribute
{
    private readonly string _displayName;

    public MyCustomDataAttribute(string displayName, string data)
    {
        _displayName = displayName;
    }

    public override IEnumerable<object[]> GetData(MethodInfo testMethod)
    {
        // Set the display name for the test case
        DisplayName = _displayName;

        yield return new object[] { data };
    }
}

[Theory]
[MyCustomData("Test case 1", "Test data 1")]
[MyCustomData("Test case 2", "Test data 2")]
public void TestMethod(string data)
{
    // Test code goes here
}
```

## Workaround

I'm aware that there is a workaround when we pass an extra parameter in the test method that doesn't participate in testing, just as a test label. However, I find it annoying.

**Comment by bradwilson** [comment]
This is already implemented for v3:

https://github.com/xunit/xunit/blob/2b1f75b320b54edc364785d5d7634deff71784f8/src/xunit.v3.core/Sdk/DataAttribute.cs#L71-L77

**Comment by voroninp** [comment]
Do I understand correctly that `TestDisplayName` completely replaces the name, and does not add just a label to the test name?

**Comment by bradwilson** [comment]
> Do I understand correctly that TestDisplayName completely replaces the name, and does not add just a label to the test name?

It replaces the use of the method name as the test name with the display name; it does not (currently) affect the printing of arguments on the test method.

**Comment by voroninp** [comment]
@bradwilson , would you consider an option for combining test name from the method and label of the input?

Something like

```
[InlineData(x1, y1, z1, Label = "Foo")]
[InlineData(X2, y2, Z2, Label = "Bar")]
Test_Xyz(a, b, c)
{
}
```

So, tests are `TestXyz_Foo` are `TestsXyz_Bar`

**Comment by bradwilson** [comment]
I like the name `Label`, as a replacement for argument lists.

* Don't set it, and you get an argument list (`Method(arg1: value1, ...)`)
* Set to a custom value to replace the argument list (`Method(CustomLabel)`)
* Set to an empty string, get no label at all (`Method`), most useful for custom display names

**Comment by bradwilson** [comment]
@voroninp This is now available in v3 `3.0.0-pre.9` https://xunit.net/docs/using-ci-builds

`Label` is settable on any data attribute as well as via `ITheoryDataRow`. From a priority perspective, setting a `Label` on a data row overrides any value set on a data attribute.
